sound/soc/sof/sof-priv.h

```diff
@@ -433,7 +433,7 @@ struct snd_sof_dev {
 	struct snd_dma_buffer dmatp;
 	int dma_trace_pages;
 	wait_queue_head_t trace_sleep;
-	u32 host_offset;
+	unsigned int host_offset;
```

Comment:
host_offset also looks strongly typed.  IIRC it's passed over IPC.

Reply:
@plbossart not directly - it's copied from a trace position IPC, yes, but then it's used in calculations in `sof_trace_avail()`, so, I don't see any reason why it must be 32 bit long.

Reply:
I think with gcc int is at least 32 bits: https://www.gnu.org/software/gnu-c-manual/gnu-c-manual.html#Data-Types To explain a bit my previous response: I meant "I don't see any reason why it must be *exactly* 32 bit long"

Reply:
having said that - I'd be happy to make it `unsigned long` or even `unsigned long long` if we suspect we actually might need to address 64 bits

Reply:
C99 remains ambiguous on the 'int' type, we should not make assumptions on compilers. Been there, done that.

Also I really don't know why you consider addresses and address offsets as generic types. To me the current encoding is a direct reflection of what the hardware expects. No less, no more, exactly 32 bits.

Reply:
@plbossart I'd bet quite a lot that we'd never run under Linux where int is 16 bits, but that's just my personal opinion anyway :-) By "addresses and address offsets" you mean only this variable or any further ones too? In general I think fixed-size variables should be used in very few cases:
1. when matching hardware registers - interrupt masks etc.
2. when those variables are a part of an ABI

We don't use `u32` / `u64` to access memory locations. We use pointers. If we have to convert them to an integer type, we use `uintptr_t`